### ISSUE TITLE
Bisq2 on Tails OS (debian) launches but does not connect to the Bitcoin network

### ISSUE THREAD

**Issue by garymansell** [bisq-network/bisq2#2715]
I have been running Bisq 1.9.15 on Tails OS using the process outlined in this guide (https://bisq.wiki/Running_Bisq_on_Tails) and the setup has been working perfectly fine and persisting the Bisq data in the Persistent storage partition of Tails USB. I just need to run the install_bisq.sh script every time Tails boots up.

I am presuming that Bisq 2 uses the same onion grater yaml settings as Bisq 1, and that I can modify the install-bisq.sh script to work the same for bisq2 and persist the data in the Tails Persistent storage partition - it seems to work and store the data in a separate location to bisq1, create the onion grater bisq1.yml file and edit the properties of the Bisq2 application with the TOR parameters OK.

This is the modified install-bisq2 script:

`#!/bin/bash
BisqInstaller=/home/amnesia/Persistent/bisq2/Bisq-2.1.0.deb
DataDirectory=/home/amnesia/Persistent/bisq2/Bisq2

echo "Install Bisq2 ..."
dpkg -i $BisqInstaller
echo "Change access rights of /var/run/tor/control.authcookie ..."
chmod o+r /var/run/tor/control.authcookie
echo "Create /etc/onion-grater.d/bisq2.yml ..."
echo "---
- apparmor-profiles:
    - '/opt/bisq2/bin/Bisq2'
  users:
    - 'amnesia'
  commands:
    AUTHCHALLENGE:
      - 'SAFECOOKIE .*'
    SETEVENTS:
      - 'CIRC ORCONN INFO NOTICE WARN ERR HS_DESC HS_DESC_CONTENT'
    GETINFO:
      - pattern: 'status/bootstrap-phase'
        response:
          - pattern: '250-status/bootstrap-phase=*'
            replacement: '250-status/bootstrap-phase=NOTICE BOOTSTRAP PROGRESS=100 TAG=done SUMMARY="Done"'
      - 'net/listeners/socks'
    ADD_ONION:
      - pattern:     'NEW:(\S+) Port=9999,(\S+)'
        replacement: 'NEW:{} Port=9999,{client-address}:{}'
      - pattern:     '(\S+):(\S+) Port=9999,(\S+)'
        replacement: '{}:{} Port=9999,{client-address}:{}'
    DEL_ONION:
      - '.+'
    HSFETCH:
      - '.+'
  events:
    CIRC:
      suppress: true
    ORCONN:
      suppress: true
    INFO:
      suppress: true
    NOTICE:
      suppress: true
    WARN:
      suppress: true
    ERR:
      suppress: true
    HS_DESC:
      response:
        - pattern:     '650 HS_DESC CREATED (\S+) (\S+) (\S+) \S+ (.+)'
          replacement: '650 HS_DESC CREATED {} {} {} redacted {}'
        - pattern:     '650 HS_DESC UPLOAD (\S+) (\S+) .*'
          replacement: '650 HS_DESC UPLOAD {} {} redacted redacted'
        - pattern:     '650 HS_DESC UPLOADED (\S+) (\S+) .+'
          replacement: '650 HS_DESC UPLOADED {} {} redacted'
        - pattern:     '650 HS_DESC REQUESTED (\S+) NO_AUTH'
          replacement: '650 HS_DESC REQUESTED {} NO_AUTH'
        - pattern:     '650 HS_DESC REQUESTED (\S+) NO_AUTH \S+ \S+'
          replacement: '650 HS_DESC REQUESTED {} NO_AUTH redacted redacted'
        - pattern:     '650 HS_DESC RECEIVED (\S+) NO_AUTH \S+ \S+'
          replacement: '650 HS_DESC RECEIVED {} NO_AUTH redacted redacted'
        - pattern:     '.*'
          replacement: ''
    HS_DESC_CONTENT:
      suppress: true" > /etc/onion-grater.d/bisq2.yml

echo "Restart onion-grater service ..."
systemctl restart onion-grater.service
echo "Edit Bisq2 executable file ..."
sed -i 's+Exec=/opt/bisq2/bin/Bisq2+Exec=/opt/bisq2/bin/Bisq2 --torControlPort 951 --torControlCookieFile=/var/run/tor/control.authcookie --torControlUseSafeCookieAuth+' /usr/share/applications/bisq2-Bisq2.desktop
echo "Redirect user data to Tails Persistent Storage ..."
ln -s $DataDirectory /home/amnesia/.local/share/Bisq2
echo "Installation complete."

`

Bisq2 starts and I get a splash screen to accept the Ts&Cs, it then sticks (at 1.25% on the progress bar) and ultimately errors trying to connect to the TOR network - when I look in the bisq2 logs (attached, there are loads of connection refused errors).

[bisq.log](https://github.com/user-attachments/files/16768230/bisq.log)

Any ideas why this is not working and how to get it working?

**Comment by garymansell** [comment]
I wonder if this is a similar issue to  #2467 ?

In which case, has the startup command (when on Tor) changed from Bisq 1 to Bisq 2, and do I need to change it in the Bisq 2 install-bisq.sh script to edit the desktop application properties?

For Bisq1, I was using this in the install-bisq.sh script:

Exec=/opt/bisq/bin/Bisq --torControlPort 951 --torControlCookieFile=/var/run/tor/control.authcookie --torControlUseSafeCookieAuth

Which I have just slightly modified for Bisq 2:

Exec=/opt/bisq2/bin/Bisq2 --torControlPort 951 --torControlCookieFile=/var/run/tor/control.authcookie --torControlUseSafeCookieAuth

Are the startup parameters for Bisq 2 the same or do they need to be different (because they quote this in the #2467 issue):

/opt/bisq2/bin/Bisq\ 2 -Dapplication.network.configByTransportType.tor.torrcOverrides.Socks5Proxy=192.0.2.1:1234

**Comment by risti1istri** [comment]
did you resolve it?

**Comment by garymansell** [comment]
Confirmed still an issue on 2.1.0 today, these are the errors in the log (there are many of them and it fails after 240 seconds trying to connect):


Sept-30 17:45:45.666 [Thread-7] INFO  b.t.c.BootstrapService: Tor bootstrap event: BootstrapEvent(progress=5, tag=conn, summary=Connecting to a relay" WARNING="Connection refused" REASON=CONNECTREFUSED COUNT=22 RECOMMENDATION=warn HOSTID="E3E42D35F801C9D5AB23584E0025D56FE2B33396" HOSTADDR="216.218.219.41:443, timestamp=2024-09-30T17:45:45.666185531Z) 
Sept-30 17:46:28.704 [Thread-7] INFO  b.t.c.BootstrapService: Tor bootstrap event: BootstrapEvent(progress=5, tag=conn, summary=Connecting to a relay" WARNING="Connection refused" REASON=CONNECTREFUSED COUNT=23 RECOMMENDATION=warn HOSTID="1A25C6358DB91342AA51720A5038B72742732498" HOSTADDR="128.31.0.39:9201, timestamp=2024-09-30T17:46:28.704352867Z)

**Comment by risti1istri** [comment]
As soon as I open BISQ2 on Tails it stops at 1.25% and after exactly 240s BISQ2 crashes completely without explanation bringing out a screen saying :

"Initialings ApplicationService Failed" 

and as soon as I close this screen another message comes out saying:

"Report bug to BISQ Developers".

 I have been trying to solve this problem for more than two days in vain without results. I looked everywhere for tutorials and explanations that could help me but unfortunately on the web there is nothing about it, the only thing I found is the problem *2715 here on GitHub. I had heard that it was necessary to install javafx but unfortunately it doesn't work I don't even know why.I'm having a lot of problems installing bisq2 as opposed to bisq1 where everything went perfectly by installing it and making it go in less than 20 min. Does anyone please know how to solve this problem???????

**Comment by HenrikJannsen** [comment]
@alvasw Any idea?

**Comment by HenrikJannsen** [comment]
@garymansell @risti1istri Can you try if there are still issues with v2.1.6?

**Comment by Jovess64** [comment]
@HenrikJannsen I am having the same issue in v2.1.6. I've just tried running as root, which causes an error initialising the socket. Here's the [bisq2-logs.zip](https://github.com/user-attachments/files/19040633/bisq2-logs.zip), if I'm interpreting it correctly it doesn't seem to be sending anything over the network at all?

**Comment by HenrikJannsen** [comment]
Have you tor installed?

**Comment by Jovess64** [comment]
Yes, tor is installed. Bisq1 works as well and the browser certainly works.
